### PR TITLE
[chore] Fix cover script generate cover report

### PR DESCRIPTION
**Pull request by heshan0131**
- generate tape test cover report with `yarn cover-tape`
- generate jest test cover report with `yanr cover-jest`
- merge json report with istanbul-merge
- generate final report with nyc
- coverall report back to 71.0% coverage
Screenshot 2024-08-20 at 4 06 22 AM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,15 +49,14 @@
...
-    "cover-tape": "nyc --reporter=json --report-dir=tape-coverage --reporter=html yarn test-tape",
-    "cover-jest": "yarn jest --coverage --reporter=html --watchAll=false",
-    "cover-copy-tape": "yarn nyc merge tape-coverage .nyc_output/coverage.json",
-    "cover-copy-jest": "yarn nyc merge jest-coverage .nyc_output/coverage.json",
-    "cover-copy": "rm -rf .nyc_output && mkdir .nyc_output && yarn cover-copy-tape && yarn cover-copy-jest",
-    "cover": "yarn cover-jest && yarn cover-tape && yarn cover-copy && nyc report --reporter=json --reporter=html --reporter=lcov",
+    "cover": "yarn cover-tape && yarn cover-jest && yarn cover-merge && yarn cover-report",
+    "cover-tape": "nyc --reporter=json --report-dir=tape-coverage --reporter=lcov yarn test-tape",
+    "cover-jest": "yarn jest --coverage --reporter=lcov --watchAll=false",
+    "cover-merge": "istanbul-merge --out=coverage/coverage-all.json tape-coverage/coverage-final.json jest-coverage/coverage-final.json",
```
Comment (ibgreen) [inline package.json:58]: Nice, didn't know merging reports was possible. Could help us introduce jest in other frameworks!

File: package.json [hunk 2]
```diff
@@ -49,15 +49,14 @@
...
     "test": "yarn test-jest && yarn test-tape",
-    "cover-tape": "nyc --reporter=json --report-dir=tape-coverage --reporter=html yarn test-tape",
-    "cover-jest": "yarn jest --coverage --reporter=html --watchAll=false",
-    "cover-copy-tape": "yarn nyc merge tape-coverage .nyc_output/coverage.json",
-    "cover-copy-jest": "yarn nyc merge jest-coverage .nyc_output/coverage.json",
-    "cover-copy": "rm -rf .nyc_output && mkdir .nyc_output && yarn cover-copy-tape && yarn cover-copy-jest",
-    "cover": "yarn cover-jest && yarn cover-tape && yarn cover-copy && nyc report --reporter=json --reporter=html --reporter=lcov",
+    "cover": "yarn cover-tape && yarn cover-jest && yarn cover-merge && yarn cover-report",
+    "cover-tape": "nyc --reporter=json --report-dir=tape-coverage --reporter=lcov yarn test-tape",
+    "cover-jest": "yarn jest --coverage --reporter=lcov --watchAll=false",
```
Comment (ibgreen) [inline package.json:57]: Not that it matters, but is `yarn` needed here?

**Review by ibgreen — APPROVED** [review]
LGTM. That is a lot of scripts in package.json...

File: package.json (hunk 1 above)
Comment (heshan0131) [reply package.json:58]: yes, jest generates its own coverage report in the same format as nyc by running `jest --coverage`, so using nyc to run jest doesn't work. But with istanbul-merge, we can merge the report generated by nyc and jest into one.

**Pull request merged**